lib/Backend/Lower.cpp

```diff
@@ -6457,9 +6457,22 @@ Lowerer::LowerAdjustObjType(IR::Instr * instrAdjustObjType)
     IR::AddrOpnd *initialTypeOpnd = instrAdjustObjType->UnlinkSrc2()->AsAddrOpnd();
     IR::RegOpnd  *baseOpnd = instrAdjustObjType->UnlinkSrc1()->AsRegOpnd();
 
-    this->GenerateAdjustBaseSlots(
+    bool adjusted = this->GenerateAdjustBaseSlots(
         instrAdjustObjType, baseOpnd, JITTypeHolder((JITType*)initialTypeOpnd->m_metadata), JITTypeHolder((JITType*)finalTypeOpnd->m_metadata));
 
+    if (adjusted)
+    {
+        // We reallocated the aux slots, so reload them if necessary.
+        StackSym * auxSlotPtrSym = baseOpnd->m_sym->GetAuxSlotPtrSym();
+        if (auxSlotPtrSym)
+        {
+            IR::Opnd *opndIndir = IR::IndirOpnd::New(baseOpnd, Js::DynamicObject::GetOffsetOfAuxSlots(), TyMachReg, this->m_func);
+            IR::RegOpnd *regOpnd = IR::RegOpnd::New(auxSlotPtrSym, TyMachReg, this->m_func);
+            regOpnd->SetIsJITOptimizedReg(true);
+            Lowerer::InsertMove(regOpnd, opndIndir, instrAdjustObjType);
```

Comment:
nit: this could be refactored to call a function on the lowerer that takes a PropertySymOpnd* and an Instr* as arguments, and replaces PropertySymOpnd::GenerateAuxSlotPtrSymLoad